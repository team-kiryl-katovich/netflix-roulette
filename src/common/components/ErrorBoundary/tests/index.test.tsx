import * as React from 'react';
import * as enzyme from 'enzyme';

import { ErrorBoundary } from '..';

describe('[COMMON] [COMPONENTS] [ERROR_BOUNDARY]', () => {
  test('should render component', () => {
    const component = enzyme.shallow(
      <ErrorBoundary>
        <div>fake children</div>
      </ErrorBoundary>,
    );
    expect(component).toMatchSnapshot();
  });

  test('should catch error', () => {
    const FakeChildren = () => <div>fake children</div>;
    const component = enzyme.mount(
      <ErrorBoundary>
        <FakeChildren />
      </ErrorBoundary>,
    );

    const fakeError = new Error('fakeError');
    const children = component.find(FakeChildren);
    children.simulateError(fakeError);

    expect(component.contains(fakeError.toString())).toBeTruthy();
  });
});
