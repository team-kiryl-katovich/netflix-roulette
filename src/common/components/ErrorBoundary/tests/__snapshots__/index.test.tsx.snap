// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMMON] [COMPONENTS] [ERROR_BOUNDARY] should render component 1`] = `
<Fragment>
  <div>
    fake children
  </div>
</Fragment>
`;
